packages/web/e2e/uploadCollection.test.ts

```diff
@@ -0,0 +1,190 @@
+import { expect } from '@playwright/test'
+import {
+  EditAlbumPage,
+  EditPlaylistPage,
+  FinishPage,
+  SelectPage
+} from './page-object-models/upload'
+import { test } from './test'
+
+test('should upload a playlist', async ({ page }) => {
+  const timestamp = Date.now()
+  const playlistName = `Test playlist ${timestamp}`
+  const playlistDescription = 'Test description'
+  const trackOneDetails = { name: `Test track 1 ${timestamp}` }
+  const trackTwoDetails = {
+    name: `Test track 2 ${timestamp}`,
+    genre: 'Alternative',
+    mood: 'Energizing',
+    tags: ['TAG3', 'TAG4']
+  }
+  const trackDetails = [trackOneDetails, trackTwoDetails]
+  const genre = 'Electronic - Progressive House'
+  const mood = 'Tender'
+  const tags = ['TAG1', 'TAG2']
+
+  await page.goto('upload')
+
+  const selectPage = new SelectPage(page)
+  await selectPage.setTracks('track.mp3', 'track-2.mp3')
+  await selectPage.setReleaseType('Playlist')
+  await selectPage.continue()
+
+  await expect(
+    page.getByRole('heading', { name: /complete your playlist/i })
+  ).toBeVisible()
+
+  const editPage = new EditPlaylistPage(page)
+  await editPage.setArtwork('track-artwork.jpeg')
+  await editPage.setTitle(playlistName)
+  await editPage.setGenre(genre)
+  await editPage.setMood(mood)
+  await editPage.setTags(tags)
+  await editPage.setDescription(playlistDescription)
+
+  for (let i = 0; i < trackDetails.length; i++) {
+    await editPage.setTrackDetails(i, trackDetails[i])
+  }
+
+  await editPage.complete()
+
+  const finishPage = new FinishPage(page, 'Playlist')
+  await finishPage.assertCompletes()
+
+  // Vist collection page
+  await page.getByRole('link', { name: /visit playlist page/i }).click()
+
+  // Assert title
+  const header = page.getByRole('heading', { name: playlistName, level: 1 })
+  await expect(header).toBeVisible()
+
+  // Assert description
+  const description = page.getByText(playlistDescription)
+  await expect(description).toBeVisible()
+
+  // Assert track list
+  const trackTable = page.getByRole('table')
+  const trackOne = trackTable.getByRole('cell', { name: trackOneDetails.name })
+  const trackTwo = trackTable.getByRole('cell', { name: trackTwoDetails.name })
+  await expect(trackOne).toBeVisible()
+  await expect(trackTwo).toBeVisible()
+
+  // Visit track 1
+  await trackOne.getByRole('link', { name: trackOneDetails.name }).click()
+
+  // Assert tagged
+  const tag1 = page.getByRole('link', { name: tags[0] })
+  const tag2 = page.getByRole('link', { name: tags[1] })
```

Comment:
super nit: I think I would prefer 
```suggestion
  const track1tag1 = page.getByRole('link', { name: tags[0] })
  const track1tag2 = page.getByRole('link', { name: tags[1] })
```
and down below
```
  const track2tag1 = page.getByRole('link', { name: tags[0] })
  const track2tag2 = page.getByRole('link', { name: tags[1] })
```
instead of 1-4